src/Storage/Concerns/HandlesDynamicFormsStorageLocal.php

```diff
@@ -0,0 +1,41 @@
+<?php
```

Comment:
Would there be a way to keep it to just the one controller?

I had aligned the controller actions in the S3 one to the resource action names (show, store, etc), but that's not a hard requirement or anything.

I know this is currently only dropping one controller into their app, but I can see somebody wanting to upload to their own server in a local dev env and S3 in production. It may be worth renaming the methods in the S3 trait, including both traits in the controller, and adjusting the routes so both sets are registered.

That would make swapping between them easier too -- only gotta implement `authorizeFileAction` & register auth middleware for one controller.

Reply:
Your suggestion of StorageRegistry might make one Controller possible, one reason I thought it had to be one way or the other was the fact that they would conflict when being registered by the form validator, I'll attempt consilidation after I get StorageRegistry change working